aiida/manage/configuration/settings.py

```diff
@@ -35,32 +36,27 @@ def create_instance_directories():
     This will create the base AiiDA directory defined by the AIIDA_CONFIG_FOLDER variable, unless it already exists.
     Subsequently, it will create the daemon directory within it and the daemon log directory.
     """
-    directory_base = os.path.expanduser(AIIDA_CONFIG_FOLDER)
-    directory_daemon = os.path.join(directory_base, DAEMON_DIR)
-    directory_daemon_log = os.path.join(directory_base, DAEMON_LOG_DIR)
+    from aiida.common import ConfigurationError
+
+    directory_base = pathlib.Path(AIIDA_CONFIG_FOLDER).expanduser()
+    directory_daemon = directory_base / DAEMON_DIR
+    directory_daemon_log = directory_base / DAEMON_LOG_DIR
 
     umask = os.umask(DEFAULT_UMASK)
+    environment_variable = os.environ.get(DEFAULT_AIIDA_PATH_VARIABLE, None)
 
     try:
-        create_directory(directory_base)
-        create_directory(directory_daemon)
-        create_directory(directory_daemon_log)
-    finally:
-        os.umask(umask)
-
+        for path in [directory_base, directory_daemon, directory_daemon_log]:
 
-def create_directory(path):
-    """Attempt to create the configuration folder at the given path skipping if it already exists
+            if path is directory_base and environment_variable and not path.exists():
+                warnings.warn(f'config directory `{path}` does not exist, creating it now.')
```

Comment:
```suggestion
                warnings.warn(f'Creating AiiDA configuration folder `{path}`.')
```
[in the docs](https://aiida.readthedocs.io/projects/aiida-core/en/latest/howto/installation.html?highlight=backup#backing-up-your-installation) it is just called the `.aiida/` folder

Reply:
Rather than updating the warning, I would update the documentation to name if the "AiiDA configuration folder", which currently just happens to be named `.aiida`. Would that be ok?

Reply:
Sure!

Reply:
Done